test/src/test/kotlin/graphql/nadel/tests/TestFixture.kt

```diff
@@ -45,6 +49,7 @@ data class ServiceCall(
     val request: Request,
     @JsonProperty("response")
     val responseJsonString: String,
+    val hydrationDetails : HydrationDetails?
```

Comment:
I'm not really sure this should be a definition inside the test spec.

Realistically we're only going to ever use this for these few tests we just added. Nobody else is going to test these details. The testing framework should be as generic as possible to avoid convoluting `EngineTests`.

I think this should've been done in a test hook e.g.

```kotlin
@UseHook
class `basic-hydration` : EngineTestHook {
    override fun makeNadel(builder: Nadel.Builder): Nadel.Builder {
        val serviceExecutionFactory = builder.serviceExecutionFactory

        return builder
            .serviceExecutionFactory(object : ServiceExecutionFactory by serviceExecutionFactory {
                override fun getServiceExecution(serviceName: String): ServiceExecution {
                    val serviceExecution = serviceExecutionFactory.getServiceExecution(serviceName)
                    return ServiceExecution { params ->
                        val details = params.hydrationDetails
                        if (details != null) {
                            assert(details.hydrationActorField.toString() == "Query.barsByIdOverall")
                            assert(details.hydrationSourceField.toString() == "Foo.bar")
                            assert(details.hydrationSourceService.name == "service1")
                        }

                        serviceExecution.execute(params)
                    }
                }
            })
    }
}
```